src/services/utilServices/AuthService.js

```diff
@@ -124,18 +126,28 @@ class AuthService {
   }
 
   async verifyOtp({ email, otp }) {
-    const isOtpValid = await this.usersService.verifyEmailOtp(email, otp)
-
-    if (!isOtpValid) {
-      throw new BadRequestError("You have entered an invalid OTP.")
-    }
-    // Create user if does not exists. Set last logged in to current time.
-    const user = await this.usersService.loginWithEmail(email)
-    const userInfo = {
-      isomerUserId: user.id,
-      email: user.email,
-    }
-    return userInfo
+    return this.usersService
+      .verifyEmailOtp(email, otp)
+      .andThen(() =>
+        ResultAsync.fromPromise(
+          this.usersService.loginWithEmail(email),
+          (error) => {
+            logger.error(
+              `Error logging user in via email: ${JSON.stringify(error)}`
+            )
+            return new DatabaseError(
+              "An error occurred when logging user in via email"
+            )
+          }
+        )
+      )
+      .map((user) => ({
+        isomerUserId: user.id,
+        email: user.email,
+      }))
+      .mapErr((error) => {
+        throw error
+      })
```

Comment:
this is kinda-self defeating but given that this is a `.js` module, `neverthrow` has limited value here anyways

Reply:
Yeah kept this mainly for backward compat